examples/features/torch_batch_process/batch_inference/compare_with_core_api/core_api_inference.py

```diff
@@ -0,0 +1,151 @@
+import torchvision.transforms as transforms
+from torch import nn
+
+import determined as det
+from determined import common, core, pytorch
+
+common.set_logger(False)
+
+
+def run_inference(
+    model: nn.Module,
+    data_loader: Any,
+    context: core.Context,
+    rank: int,
+    skip: int,
+    per_worker_iterate_length: int,
+) -> None:
+    model.eval()
+    records_processed = 0
+
+    inference_output_dir = "inference_out/"
+
+    # The first worker will create it, and exist_ok option makes sure subsequent workers
+    # do not run into error
+    pathlib.Path.mkdir(pathlib.Path(inference_output_dir), parents=True, exist_ok=True)
+
+    dataloader_iterator = iter(data_loader)
+
+    with torch.no_grad():
+        for batch_idx in range(skip, per_worker_iterate_length):
+            X = next(dataloader_iterator, None)
+            logging.info(f"Working on batch is {batch_idx}")
+            if X is not None:
+                data, label = X
+                output = model(data)
+                preds = output.argmax(dim=1, keepdim=True)
+
+                file_name = f"inference_out_{rank}_{batch_idx}.json"
+
+                output = []
+
+                for pred in preds:
+                    output.append(pred[0].item())
+
+                with open(
+                    os.path.join(inference_output_dir, f"{file_name}"),
+                    "w",
+                ) as f:
+                    json.dump({"predictions": output}, f)
+
+                # After each batch, synchronize and update number of catches completed
+                if context.distributed.rank == 0:
+                    work_completed_this_round = sum(context.distributed.gather(len(data)))
+                    records_processed += work_completed_this_round
+                    checkpoint_metadata = {
+                        "steps_completed": batch_idx,
+                    }
+                    with context.checkpoint.store_path(checkpoint_metadata) as (path, uuid):
+                        with open(os.path.join(path, "batch_completed.json"), "w") as file_obj:
+                            json.dump({"batch_completed": batch_idx}, file_obj)
+                else:
+                    context.distributed.gather(len(data))
+
+                if context.preempt.should_preempt():
+                    return
+
+
+def _get_data_loader(batch_size: int, total_worker: int, rank: int, skip: int) -> [Any, int]:
+    transform = transforms.Compose(
+        [transforms.ToTensor(), transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))]
+    )
+    with filelock.FileLock(os.path.join("/tmp", "inference.lock")):
+        inference_data = tv.datasets.CIFAR10(
+            root="/data", train=False, download=True, transform=transform
+        )
+    dataloader = pytorch.DataLoader(
+        dataset=inference_data,
+        batch_size=batch_size,
+        shuffle=False,
+    ).get_data_loader(repeat=False, skip=skip, num_replicas=total_worker, rank=rank)
```

Comment:
get_data_loader is not a public API, although it is so old we didn't prefix it with an underscore.

Wouldn't hurt to add the underscore today though, I think.

Reply:
Got it, I will remove reference to this function.

Reply:
yeah, sorry I wasn't trying to imply that I wanted you to include the rename in this pr, by any means.